src/app/core/user-module/user.service.ts

```diff
@@ -29,9 +29,8 @@ export class UserService {
       })
       .pipe(catchError(this.handleError))
       .toPromise()
-      .then((val: any) => {
-        const arr = val;
-        return arr.map((user) => new User(user));
+      .then((res: {users: User[], total: number}) => {
```

Comment:
🎨

Reply:
does this fix work the same as changing it to `arr.users.map`?